fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1065,16 +1067,15 @@ private Subspace indexBuildLockSubspace(@Nonnull FDBRecordStore store, @Nonnull
         if (indexStatePrecondition == IndexStatePrecondition.ERROR_IF_DISABLED_CONTINUE_IF_WRITE_ONLY) {
             return doBuildIndexAsync(markReadable);
         }
-        return getRunner().runAsync(context -> openRecordStore(context).thenApply(store -> {
+        return getRunner().runAsync(context -> openRecordStore(context).thenCompose(store -> {
             IndexState indexState = store.getIndexState(index);
             if (shouldBuildIndex(indexState, indexStatePrecondition)) {
                 if (!shouldNotClearExistingIndexEntries(indexState, indexStatePrecondition)) {
                     store.clearIndexData(index);
                 }
-                store.markIndexWriteOnly(index);
-                return true;
+                return store.markIndexWriteOnly(index).thenApply(vignore -> true);
```

Comment:
- Is there any particular reason why it is named `markIndexWriteOnly` rather than `markIndexWriteOnlyAsync` in `FDBRecordStore`?
- I'd like to add a logging to show all the choices here (use synchronized session or not, clear the entries or not, change the state or not, need build or not). Is there a good way to combine them in **one** log?

Reply:
Yeah. Not suffixing `Async` will also confuse the `checkIfBlockingInFuture` I think.

Reply:
I filed #775 about this.